src/System.CommandLine/Parsing/StringExtensions.cs

```diff
@@ -405,16 +410,17 @@ static IEnumerable<string> SplitLine(string line)
             }
         }
 
-        private static Dictionary<string, CliToken> ValidTokens(this CliCommand command, IReadOnlyList<CliDirective>? directives)
+        private static Dictionary<string, CliToken> ValidTokens(this CliCommand command)
         {
             Dictionary<string, CliToken> tokens = new(StringComparer.Ordinal);
 
-            if (directives is not null)
+            if (command is CliRootCommand { Directives: IList<CliDirective> directives })
```

Comment:
again subjective on style: what do we benefit from using this syntax here?

```cs            
if (command is CliRootCommand { Directives: IList<CliDirective> directives })
{
    for (int i = 0; i < directives.Count; i++)
```

```cs
if (command is CliRootCommand rootCommand)
{
    for (int i = 0; i < rootCommand.Directives.Count; i++)
```

Reply:
This avoids an intermediate assignment of a `CliRootCommand` variable when all we need from it is the directives:

```csharp
if (command is CliRootCommand { Directives: IList<CliDirective> directives })
```

I don't know if this makes any difference from a performance perspective but I find it (subjectively, again) to be less cluttered.